engine/src/flutter/shell/platform/windows/host_window.h

```diff
@@ -85,27 +115,63 @@ class HostWindow {
     MONITORINFO monitor_info;
   };
 
-  HostWindow(WindowManager* controller,
+  HostWindow(WindowManager* window_manager,
              FlutterWindowsEngine* engine,
              WindowArchetype archetype,
-             std::unique_ptr<FlutterWindowsViewController> view_controller,
-             const BoxConstraints& constraints,
-             HWND hwnd);
+             DWORD window_style,
+             DWORD extended_window_style,
+             const BoxConstraints& box_constraints,
+             Rect const initial_window_rect,
+             LPCWSTR title,
+             HWND owner_window);
+
+  // Calculates the required window size, in physical coordinates, to
+  // accommodate the given |client_size|, in logical coordinates, constrained by
+  // optional |smallest| and |biggest|, for a window with the specified
+  // |window_style| and |extended_window_style|. If |owner_hwnd| is not null,
+  // the DPI of the display with the largest area of intersection with
+  // |owner_hwnd| is used for the calculation; otherwise, the primary display's
+  // DPI is used. The resulting size includes window borders, non-client areas,
+  // and drop shadows. On error, returns std::nullopt and logs an error message.
+  static std::optional<Size> GetWindowSizeForClientSize(
+      WindowsProcTable const& win32,
+      Size const& client_size,
+      std::optional<Size> smallest,
+      std::optional<Size> biggest,
+      DWORD window_style,
+      DWORD extended_window_style,
+      HWND owner_hwnd);
+
+  // Processes and routes salient window messages for mouse handling,
+  // size change and DPI. Delegates handling of these to member overloads that
+  // inheriting classes can handle.
+  virtual LRESULT HandleMessage(HWND hwnd,
+                                UINT message,
+                                WPARAM wparam,
+                                LPARAM lparam);
 
   // Sets the focus to the child view window of |window|.
-  static void FocusViewOf(HostWindow* window);
+  static void FocusRootViewOf(HostWindow* window);
+
+  // Enables or disables this window and all its descendants.
+  void EnableWindowAndDescendants(bool enable);
```

Comment:
Ideally this method would align with its disable counterpart. What do you think of naming the methods `EnableRecursively`?